### PR TITLE
CEDS 2102 - pre-populate query ucr in journeys

### PR DESCRIPTION
**Pull request by SteveSugden**
This draft PR completes the CEDS-2102 spike - pre-populating the queried ucr (ducr/mucr) into the various journeys.

If we are happy with this approach and can complete with tests.

### CODE REVIEW

File: app/models/UcrBlock.scala [hunk 1]
```diff
@@ -22,4 +22,7 @@ case class UcrBlock(ucr: String, ucrType: String)
 
 object UcrBlock {
   implicit val format = Json.format[UcrBlock]
+
+  val ducr = "D"
+  val mucr = "M"
```
Comment (MaciejRewera) [inline app/models/UcrBlock.scala]: Maybe better call them `DucrType` and `MucrType` ?

Comment (SteveSugden) [reply app/models/UcrBlock.scala]: Can do.  And I don't think I've used them consistently - there are still some hard-coded "D" and "M" out there.

Doing this exercise highlighted how many times we have created constants/objects/enums to represent ducr or mucr in the product though!  :-)

File: app/models/cache/Answers.scala [hunk 2]
```diff
@@ -57,6 +59,15 @@ case class ArrivalAnswers(
 
 object ArrivalAnswers {
   implicit val format: Format[ArrivalAnswers] = Json.format[ArrivalAnswers]
+
+  def apply(queryUcr: Option[UcrBlock]): ArrivalAnswers = queryUcr match {
+    case Some(ucrBlock) if ucrBlock.ucrType.equals(mucr) =>
+      ArrivalAnswers(consignmentReferences = Some(ConsignmentReferences(ConsignmentReferenceType.M, ucrBlock.ucr)))
```
Comment (MaciejRewera) [inline app/models/cache/Answers.scala]: Could you please use parameter names for `ConsignmentReferences` ?

Comment (SteveSugden) [reply app/models/cache/Answers.scala]: You mean `ConsignmentReferences(reference = ConsignmentReferenceType.M, referenceValue = ucrBlock.ucr))` ?  Is that necessary for clarity?

Comment (PatrykRudnicki) [reply app/models/cache/Answers.scala]: I think it's not necessary in cases where we're filling all arguments and additionally those arguments are really good described. I mean `ConsignmentReferenceType.M` `ucrBlock.ucr`

Comment (PatrykRudnicki) [reply app/models/cache/Answers.scala]: Why you're not using `UcrBlock.ducr`, but `ConsignmentReferenceType.D` ?

Comment (SteveSugden) [reply app/models/cache/Answers.scala]: Don't understand . 
UcrBlock.mucr => ConsignmentReferenceType.M
UcrBlock.ducr => ConsignmentReferenceType.D

**Comment by SteveSugden** [comment]
Thanks for the comments.  I've demo'd to Ryan and he's happy to proceed so I'll work this up into a proper PR with tests.

File: app/models/cache/Answers.scala (hunk 2 above)
Comment (PatrykRudnicki) [reply app/models/cache/Answers.scala]: My fault, misunderstood. It's okay. I thought that UcrBlock.mucr has the same value like ConsignmentReferenceType.M

**Pull request closed without merging**